### PR TITLE
ES|QL: Fix error messages for non-snapshot tests

### PR DESCRIPTION
**Pull request by astefan**
Following https://github.com/elastic/elasticsearch/pull/111995 some error messages have changed for those features/commands that are also protected at the parser level. These only happen to be tested when `-Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false"` is used and, apparently, they were never caught by CI.

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java [hunk 1]
```diff
@@ -1904,9 +1904,9 @@ public void testLookup() {
             | RENAME languages AS int
             | LOOKUP int_number_names ON int
             """;
-        if (Build.current().isProductionRelease()) {
```
Comment (astefan) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java:1907]: We seem to sometimes (rarely) use this check, while we _mostly_ use the `isSnapshot()` one. I _think_ there is a small difference between them and here I tried to have the same approach consistent.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java [hunk 2]
```diff
@@ -4431,11 +4431,16 @@ public void testReplaceSortByExpressionsWithStats() {
...
-        var plan = optimizedPlan("""
+        var query = """
             FROM test
             | INLINESTATS c = COUNT(salary) by emp_no % 2
-            """);
-
+            """;
+        if (Build.current().isSnapshot() == false) {
+            var e = expectThrows(ParsingException.class, () -> analyze(query));
+            assertThat(e.getMessage(), containsString("line 2:3: mismatched input 'INLINESTATS' expecting {"));
```
Comment (astefan) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java:4440]: `inlinestats` is special in that it's behind a feature_flag, which is not available/used in the parser checks.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java [hunk 3]
```diff
@@ -34,24 +35,31 @@
 import static org.hamcrest.Matchers.startsWith;
 
 public class CaseTests extends AbstractScalarFunctionTestCase {
-    private static final List<DataType> TYPES = List.of(
-        DataType.KEYWORD,
-        DataType.TEXT,
-        DataType.BOOLEAN,
-        DataType.DATETIME,
-        DataType.DATE_NANOS,
```
Comment (astefan) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java:42]: This one caught me by surprise...

Comment (nik9000) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java:42]: Oh neat! if we're not on snapshot we can't use DATE_NANOS.....

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
I've added the magic `test-release` tag which'll run a release test on the build and should catch any other release-related failures....

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java (hunk 3 above)
Comment (not-napoleon) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java:42]: Huh.  I wonder why we didn't catch a problem here when I added the data type and feature flag.  I did put the `test-release` tag on https://github.com/elastic/elasticsearch/pull/110205

**Review by not-napoleon — APPROVED** [review]
I think this is fine for the moment, but I feel like we are over-using the build checks (either `isSnapshot` or `isProductionRelease`).  As a team we agreed (in 2019) to adopt feature flag based development, and that's what we should be doing here instead of checking the build status.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by martijnvg — APPROVED** [review]
LGTM - Thanks for addressing these release test failures.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/release-tests

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/release-tests

**Pull request merged**